This project / notebook is created for the Toronto neighborhood assignment

In [43]:
import numpy as np 

import pandas as pd

import json # library to handle JSON files

#!conda install -c conda-forge geopy --yes # comment after first run
from geopy.geocoders import Nominatim # convert an address into latitude and longitude values

import requests # library to handle requests
from pandas.io.json import json_normalize # tranform JSON file into a pandas dataframe

# Matplotlib and associated plotting modules
import matplotlib.cm as cm
import matplotlib.colors as colors

# import k-means from clustering stage
from sklearn.cluster import KMeans

#!conda install -c conda-forge folium=0.5.0 --yes # comment after first run for foursquare apis
#import folium # map rendering library

print('Libraries imported.')

Libraries imported.


In [45]:
# reading the wikipedia page and getting the  postal codes of canada data into a dataframe

from pandas.io.html import read_html
page = 'https://en.wikipedia.org/wiki/List_of_postal_codes_of_Canada:_M'
df = read_html(page, attrs={"class":"wikitable"})[0]
df.shape

(180, 3)

At this point, we have read the wikipedia page and got the table into the dataframe df.
the code in the next cell removes the rows which have Borough as Not assigned and the remaining rows are stored in df itself.

In [46]:
df = df[df['Borough'] != "Not assigned"]
df.head()

Postal Code           Borough                                Neighbourhood
2         M3A        North York                                    Parkwoods
3         M4A        North York                             Victoria Village
4         M5A  Downtown Toronto                    Regent Park, Harbourfront
5         M6A        North York             Lawrence Manor, Lawrence Heights
6         M7A  Downtown Toronto  Queen's Park, Ontario Provincial Government

The code in the below cell checks if there are any rows which have Neighbourhood as Not assigned.  Please note that there are zero rows and hence we are good to go!!!

In [47]:
x = df[df['Neighbourhood'] == "Not assigned"]
x.shape

(0, 3)

In [48]:
df.shape

(103, 3)

The above code tells us that there are 103 rows of data with the 3 columns Postal code, Borough and Neighbourhood in the dataframe df.